drivers/hmis/app/graphql/mutations/update_form_definition.rb

```diff
@@ -35,5 +36,41 @@ def resolve(id:, input:)
         { errors: errors }
       end
     end
+
+    private
+
+    def convert_form_definition(definition)
+      {
+        item: definition['item'].map do |i|
+          convert_form_item(i)
+        end,
+      }
+    end
+
+    def convert_form_item(item)
+      converted = basic_convert(item)
+      converted['mapping'] = basic_convert(item['mapping']) if item['mapping']
```

Comment:
This needs to account for all other attributes that can contain nested objects, so they get transformed correctly. This should be all of them:

mapping
bounds
pick_list_options
enable_when
initial
autofill_values
autofill_values => autofill_when

Reply:
I ended up feeling like a transformation function that's totally agnostic to the specific keys it needs to transform was a bit more graceful, because we separately check the input against the expected json schema. Let me know if you think this looks ok!

Reply:
looks good to me. I think there was just the issue of handling primitive arrays. I'm also not sure how it will handle empty arrays/objects, and whether that matters.

Reply:
I think empty arrays/objects should be okay, the loop on the array (or the hash's keys) will just do nothing